username.rb

```diff
@@ -2,13 +2,110 @@
 # Run `rspec /spec/username_spec.rb` to get started.
 
 def format_name(first, last)
-  nil
+  if last.length < 1 || first.length < 1
+    nil
+  else
+    first_name = ''
+    regex = /[a-zA-Z]/
+    first.split("").each do |first_char|
+      if first_char =~ regex
+        first_name += first_char
+      end
+    end
+    last_name = ''
+    last.split("").each do |character|
+      if character =~ regex
+        last_name += character
+      end
+    end
+    first_name[0].downcase + last_name.downcase.delete(" ")
+  end
+
+end
+
+def format_year(year)
+  string_year = year.to_s.split("")
+  if string_year.length != 4
+  else
+    string_year[2] + string_year[3]
+  end
+end
+
+def build_username(first, last, year, level=0)
+  level = level.floor
+  priv = ""
+  if level > 3
+    level = 3
+  elsif level < 0
+    level = 0
+  end
+  if level == 0
+    priv = ""
+  elsif level == 1
+    priv = "seller-"
+  elsif level == 2
+    priv = "manager-"
+  elsif level == 3
+    priv = "admin-"
+  else
+    priv = ""
+  end
+  string_year = year.to_s.split("")
+  priv + first.split("")[0].downcase+last.downcase + string_year[2] + string_year[3]
+end
+
+def check_privilege(level=0)
+  level = level.floor
+  if level > 3
+    level = 3
+  elsif level < 0
+    level = 0
+  end
+  if level == 0
+    "user"
+  elsif level == 1
+    "seller"
+  elsif level == 2
+    "manager"
+  elsif level == 3
+    "admin"
+  else
+    "user"
+  end
 end
 
-def format_year
-  nil
+def user_type_prefix(level=0)
```

Comment:
You're writing the same code three times here - how can you refactor this so that the code is DRY?